include/silo/common/nucleotide_symbols.h

```diff
@@ -31,37 +33,46 @@ enum class NUCLEOTIDE_SYMBOL : char {
    N,    // any base
 };
 
-static constexpr uint32_t NUC_SYMBOL_COUNT = static_cast<uint32_t>(NUCLEOTIDE_SYMBOL::N) + 1;
+template <>
+class Util<NUCLEOTIDE_SYMBOL> {
+  public:
+   static constexpr uint32_t count = static_cast<uint32_t>(NUCLEOTIDE_SYMBOL::N) + 1;
 
-static constexpr std::array<NUCLEOTIDE_SYMBOL, NUC_SYMBOL_COUNT> NUC_SYMBOLS{
-   NUCLEOTIDE_SYMBOL::GAP,
-   NUCLEOTIDE_SYMBOL::A,
-   NUCLEOTIDE_SYMBOL::C,
-   NUCLEOTIDE_SYMBOL::G,
-   NUCLEOTIDE_SYMBOL::T,
-   NUCLEOTIDE_SYMBOL::R,
-   NUCLEOTIDE_SYMBOL::Y,
-   NUCLEOTIDE_SYMBOL::S,
-   NUCLEOTIDE_SYMBOL::W,
-   NUCLEOTIDE_SYMBOL::K,
-   NUCLEOTIDE_SYMBOL::M,
-   NUCLEOTIDE_SYMBOL::B,
-   NUCLEOTIDE_SYMBOL::D,
-   NUCLEOTIDE_SYMBOL::H,
-   NUCLEOTIDE_SYMBOL::V,
-   NUCLEOTIDE_SYMBOL::N,
-};
+   static constexpr std::string_view symbol_name = "Nucleotide";
+   static constexpr std::string_view symbol_name_lower_case = "nucleotide";
+   static constexpr std::string_view symbol_name_upper_case = "NUCLEOTIDE";
+   static constexpr std::string_view symbol_name_short = "NUC";
```

Comment:
Please check case style of constexpr values in this file. They should be UPPER_CASE.